### PR TITLE
Fix incremental project generation swift debugging

### PR DESCRIPTION
**Pull request by yongjincho92**
Incremental project generation mode was not setting swift debug settings when top level target (e.g. ios application) is an Obj-c target. This change should fix the issue. 

Before : lldb > `settings show target.swift-extra-clang-flags` is empty 
After : lldb > `settings show target.swift-extra-clang-flags` is not empty

### CODE REVIEW

File: tools/generators/target_build_settings/src/Generator/ProcessSwiftArgs.swift [hunk 1]
```diff
@@ -114,9 +158,8 @@ extension Generator.ProcessSwiftArgs {
         guard let tool = try await iterator.next(),
               tool != Generator.argsSeparator
         else {
-            return (false, [], [], [])
+            return (false, [], [], [], [])
         }
-        _ = try await iterator.next()
```
Comment (brentleyjones) [inline tools/generators/target_build_settings/src/Generator/ProcessSwiftArgs.swift:119]: Why was this removed?

Comment (yongjincho92) [reply tools/generators/target_build_settings/src/Generator/ProcessSwiftArgs.swift:119]: oops, removed by mistake. let me put this back

**Review by brentleyjones — APPROVED** [review]
